### ISSUE TITLE
MPI suffixes in shared library rpmdeps are not handled correctly, giving bogus errors

### ISSUE THREAD

**Issue by jwakely** [rpminspect/rpminspect#799]
See https://fedora.softwarefactory-project.io/rpminspect-report/?url=https://centos.softwarefactory-project.io/logs//2/2/e3e0e446eddd9ae6fc21a11e533f535fee1a8b6e/check/rpm-rpminspect/d588e2a/result.json where we have **rpmdeps** results like:

> Subpackage boost-graph-mpich on x86_64 carries 'Requires: libboost_mpi.so.1.75.0()(64bit)(mpich-x86_64)' which comes from subpackage boost-openmpi but does not carry an explicit package version requirement. Please add 'Requires: boost-openmpi = %{version}-%{release}' to the spec file to avoid the need to test interoperability between various combinations of old and new subpackages.
>
>    Remedy: Add the indicated explicit Requires to the spec file for the named subpackage. Subpackages depending on shared libraries in another subpackage must carry an explicit 'Requires: SUBPACKAGE_NAME = %{version}-%{release}' in the spec file.

This is incorrect, the `boost-openmpi` package does **not** provide that, it provides `libboost_mpi.so.1.75.0()(64bit)(openmpi-x86_64)`, note the different suffix: (openmpi-x86_64) vs (mpich-x86_64)

It seems that rpminspect is ignoring these additional MPI suffixes and then giving bogus errors and suggesting bogus remedies.

The next **rpmdeps** error is:

> Multiple subpackages provide 'Requires: libboost_mpi.so.1.75.0()(64bit)(mpich-x86_64)': boost-mpich, boost-openmpi
>
>    Remedy: Check subpackage %files sections and explicit Provides statements. Only one subpackage should provide a given shared library. Shared library names are automatically added as Provides, so there is no need to specify them in the spec file but you do need to make sure only one subpackage is packaging up the shared library in question.

This is also bogus, for exactly the same reason. `boost-mpich` provides that lib with a (mpich-x86_64) suffix and boost-openmpi provides it with a (openmpi-x86_64) suffix. That's not the same thing.

The other seven rpmdeps errors in that report are due to the same issue.

**Comment by dcantrell** [comment]
The trailing qualifiers like (x86-64) and (mpich-x86_64) should be excluded for the purposes of comparisons at that point, but I obviously have a mistake here.  Digging in to it.

**Comment by dcantrell** [comment]
Yeah, that's what I'm seeing.  OK, so what I can do for this is continuing striking (arch) but leaving the other qualifiers in place.  I'll see what I can work up today.

**Comment by dcantrell** [comment]
So I have a patch now for the rpmdeps inspection that more properly handles the ISA substrings that end up in dependency requirement strings.  That reduces the number of errors to 11 per architecture for the boost package.

Looking at the package itself, I see some subpackages carry explicit Requires and others are missing them.  rpminspect is reporting on the ones that are missing those.

PR posting soon.
